README.md

```diff
@@ -201,6 +201,219 @@ hints and first-parties can use `Feature-Policy` to decide which third-parties t
+redirect would be required here in case where the hints are not present in a
+browser that supports them.  There's [ongoing
+work](https://github.com/WICG/client-hints-infrastructure/issues/16) to
+eliminate that extra step.
+
+Third-party origins that need to perform such adaptation would need
+[delegation](https://github.com/WICG/client-hints-infrastructure#cross-origin-hint-delegation)
+from the top-level origin. The top-level origin would need to opt-in using
+`Accept-CH`, as well as add `Feature-Policy` headers that delegate those hints
+to the third-party origin.
+
+### OS specific styles
+Some sites may wish to tailor their interfaces to match the user's OS. While
+progressive enhancement is likely to be a better path here (e.g. through the
+application of different button styles using script), there may be cases where
+folks would wish to deliver tailored inline styles based on the platform and
+platform version.
+
+Those cases are very similar to the case discussed above (in "Low-powered
+devices"), only with the `UA-Platform` and `UA-Platform-Version` hints.
+
+### OS integration
+Similarly, some sites would want to change links to OS specific ones (e.g.
+[Android intent
+links](https://developer.chrome.com/multidevice/android/intents)).  While,
+again, progressive enhancement can be used to modify those links using script,
+rather than bake them into the HTML, some sites may prefer server-side
+adaptation.
+
+Again, like the "OS specific styles" case, they'd need to use the platform and
+platform version hints to do so.
+
+### Browser and OS specific experiments
+Some servers may like to limit their multi variant experimentation to specific
+browsers, specific platforms or specific versions of any of the above.  For
+experiments that are limited to browser and version, those sites can use the
+`Sec-CH-UA` values sent by default on requests.  If they require platform and
+its version, they'd have to opt-in for those hints, or use client-side scripts
+to control the experimentation.
+
+## User login notification
+Many sites, especially security sensitive ones, like to notify their users when
+log-in from a new device happens. That enables users to be aware of those
+logins, and take action in case it's not a login that's done by them or on
+their behalf.
+
+For those notifications to be meaningful, sites need to recognize and
+communicate the commercial brand of the browser to the user. These messages
+often also include the platform and its version in order to make sure the user
+knows which device is in question.
+
+Since such messaging doesn't require any server-side adaptation, it's better
+for this case to use the `getUserAgent` method in order to retrieve the
+  required information.
+
+## Conversion modeling
+Some machine learning models use various details from the `User-Agent` string
+in order to estimate various things about users of those user agents.  Similar
```

Comment:
This seems very close to fingerprinting...

Reply:
It's definitely using a lot of high entropy signals in ways that are similar to fingerprinting. At the same time, I think it's distinct from user tracking or the spam avoidance cases.